doc/embedsource.md

```diff
@@ -0,0 +1,69 @@
+# Using picolibc as an embedded source
+
+For projects where deterministic builds, reproduceability and traceability is important, it is possible to embed the entire picolibc library as a meson subproject into the source directory of the master project. The benefit is that git submodules and meson subprojects work nicely together, and it is possible to lock down the libc version together with the source tree.
+
+For more information on the meson subproject feature, refer to the documentation at : https://mesonbuild.com/Subprojects.html
+
+## Adding picolibc as a subproject
+
+First you need to add picolibc as a git submodule:
+
+  ```
+  mkdir subprojects
+  git submodule add https://github.com/keith-packard/picolibc subprojects/picolibc
+  ```
+
+Your source direcoty could now look something like this:
+
+  ```
+  subprojects/
+    picolibc/
+    otherlib/
+  src/
+    main.c
+  meson.build
+  ```
+
+## Add dependency to meson.build
+
+First of all, when using picolibc as an embedded source, you should not specify -specs=picolibc.specs in the c_args.
+
+Next you will need to include the dependency in your meson.build file:
+
+  ```
+  dependencies += dependency('picolibc', fallback: ['picolibc', 'picolibc_dep'])
+  ```
+  
+In order to avoid linker conflicts with any compiler bundled c-library, use the following link flags (also see the warning below):
+
+  ```
+  link_args += '-nolibc'
+  link_args += '-lgcc'
+  ```
+  
+Finally use the dependency and linker arguments on your executable
+
+  ```
+  exec = executable('myproject.elf', sources, dependencies: dependencies, link_args: link_args)
+  ```
+
+## Warning: Remove bundled libc (newlib)
```

Comment:
See above -- creating a custom .specs file for the embedded library would avoid this issue